handout/architect/architect-guide-source-code-to-working-software.md

```diff
@@ -0,0 +1,128 @@
+      need for a new layer of abstraction 
+- Note that we're working with source code that is compiled, ie our code is not
+literally passed to the runtime "word for word"
+- Describe at an abstract level the complete process of turning our app code 
+into at a high level and how our commands work their way through the stack
+
+This would ideally take about five minutes as the audience should know this
+already
+
+
+### The Towering Web and Other Stacks
+
+_or why are we even using this stuff?_
+
+- Note the obvious fact that we could just write something like C but also note
+that it's much harder to share (provide) software to people or machines with
+traditional "Native Code".
+- Note that the Java world has "solved" this portability problem another way but
+also note that Java does not solve the _distribution_ aspect of the problem
+- Note that the web existed and had some basic capabilities and that it
+_evolved_ into an application stack that was not just portable in the 
+traditional sense but also inherently distributable, and fortunately ubiquitous.
+
+This would ideally be about three minutes since it could be distilled in a
+handful of sentences.
+
+
+### The Problem Angular and TypeScript Solve
+
+- The browser and specifically the DOM were designed to render _documents_ not
+programs but documents
+- Working _efficiently_ with the DOM to replicate "native" software look/feel
+has challenges
+- Efficiently organizing source code in a language that was created in ten days
+is non-trivial, consequently harder to maintain, teach, etc
+- Efficiently organizing the aforementioned source code along with the two DOM
+languages - HTML and CSS - is also challenging
+- Maintaining the tools necessary to orchestrate all of these things is yet
+another task unto itself
+
+This would also ideally be about five minutes
+
+
+## Act II - Making Things Happen
+
+### From Code To Runnable Code
+
+Starting sample code: _Working angular environment with zero moving parts a
+single component "Hello World" will do_
+
+- Quickly walk through the sample code to highlight all of the TypeScript code
+and the references to Angular library code
+- Run TypeScript to produce a regular transpilation (individual files, not a 
+bundle)
+- Compare the input source to the ES5 code that will actually be executed and
+note the decorators are just functions but don't get into details
+
+This would ideally be a five minute procedure
+
+
+### From Runnable Code To The User
+
+- Describe at a very specific but high level the physical networking and wires, 
+TCP, HTTP, DNS
+- Enter the web server
+- Run the code with live server and marvel at "Hello World"
+- Build/Run/Execute a specially crafted Hello World that depends on an Angular
+lib marked with `debugger` statements (to facilitate fast iteration)
+- During this debugger task stop at the specific phases of what is going on, 
+do not get bogged down into the dark corners
```

Comment:
OK - what specific phases are you going to show? (I think this is a large part of the meat of the article.)